emscripten.py

```diff
@@ -297,6 +297,11 @@ def emscript(in_wasm, out_wasm, outfile_js, memfile):
 
   metadata = finalize_wasm(in_wasm, out_wasm, memfile)
 
+  if '__main_argc_argv' in metadata['exports']:
+    settings.MANGLED_MAIN = 1
+    metadata['exports'].remove('__main_argc_argv')
+    metadata['exports'].append('main')
```

Comment:
How does this work? That is, if there is an export in the actual wasm of the long name, how can we remove it from the metadata here and replace it with the short name? Won't that confuse the JS?

Reply:
Hopefully the comment I added helps explain..